src/program.cpp

```diff
@@ -507,10 +507,8 @@ std::vector<argument> generic_eval(const module* mod,
                 }));
         }
         assert(results.find(ins) != results.end());
-        if(not ins->get_shape().any_of_dynamic())
-        {
-            assert(results.at(ins).get_shape() == ins->get_shape());
-        }
+        assert(ins->get_shape().any_of_dynamic() or
```

Comment:
why drop not here?

Reply:
So the `assert` will only assert `results.at(ins).get_shape() == ins->get_shape()` when `ins->get_shape().any_of_dynamic()` is false, to match the previous behavior.

Reply:
AH you're trying to save a compare. Got it.